src/Violations.elm

```diff
@@ -0,0 +1,225 @@
+                                    ]
+                                )
+                            |> Maybe.withDefault []
+                            |> List.append [ Check.todoComment comment ]
+                            |> List.filterMap identity
+
+            Nothing ->
+                case entity.eType of
+                    Models.Function params ->
+                        params
+                            |> List.map Check.NotAnnotatedArgument
+                            |> (::) Check.NoEntityComment
+
+                    Models.Record fields ->
+                        fields
+                            |> List.map Check.NotAnnotatedArgument
+                            |> (::) Check.NoEntityComment
+
+                    _ ->
+                        [ Check.NoEntityComment ]
+        )
+            |> filterIgnored ignored
+
+
+{-| Computes the failed checks for a dangling comment.
+
+  - `ignored` -- the list of the checks not to be performed.
+
+-}
+dangling : Models.Comment -> List Check.Type -> List Check.Type
+dangling ( _, comment ) ignored =
+    let
+        commentLine =
+            if String.startsWith "--" comment then
+                comment
+
+            else
+                String.split "\n" comment
+                    |> List.head
+                    |> Maybe.withDefault ""
+    in
+    [ Check.startingCapitalized commentLine
+    , Check.startingSpace commentLine
+    , Check.endingPeriod commentLine
+    , Check.commentType commentLine
+    , Check.todoComment commentLine
+    ]
+        |> List.filterMap identity
+        |> filterIgnored ignored
+
+
+{-| Computes the failed checks for a top-level comment (or lack thereof).
+
+  - `ignored` -- the list of the checks not to be performed.
+
+-}
+topLevel : Maybe Models.Comment -> List Check.Type -> List Check.Type
+topLevel maybeComment ignored =
+    (case maybeComment of
+        Just ( _, comment ) ->
+            [ Check.startingSpace comment
+            , Check.todoComment comment
+            ]
+                |> List.filterMap identity
+
+        Nothing ->
+            [ Check.NoTopLevelComment ]
+    )
+        |> filterIgnored ignored
+
+
+{-| Contains a matcher for a documentation argument check.
+
+It matches arguments documented like:
+"\* ´argname´ --", or
+"\* ´argname´ &mdash;".
+
+  - ´hello´ -- some explanation
+
+-}
+argumentRegex : Regex.Regex
+argumentRegex =
+    Regex.regex "\\* ´([a-zA-Z0-9]+)´ (--|&mdash;)"
+
+
+{-| Parses the documented arguments or fields from a comment.
+-}
+parseArguments : String -> List String
+parseArguments comment =
+    let
+        matchToArgumentName match =
+            match.submatches
+                |> List.map (Maybe.withDefault "")
+                |> List.head
+                |> Maybe.withDefault ""
+    in
+    comment
+        |> Regex.find Regex.All argumentRegex
+        |> List.map matchToArgumentName
+
+
+{-| Returns the difference of the second list and the first.
```

Comment:
This is a bit non-informative comment -- you called the function "filterIgnored" for a reason (otherwise you would have called it diff or something like that). I'd rather describe why this function exists (Filters the ignored checks from the list of all checks?) rather than what it does.